File: src/main/java/com/holmraven/chickensshed/ChickensShed.java
package com.holmraven.chickensshed;

import com.holmraven.chickensshed.client.GUIConfigHandler;
import com.holmraven.chickensshed.config.ConfigHandler;
import me.shedaniel.autoconfig.AutoConfig;
import me.shedaniel.autoconfig.serializer.JanksonConfigSerializer;
import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.entity.animal.Chicken;
import net.minecraft.world.item.Items;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.event.entity.living.LivingEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

@Mod(ChickensShed.MODID)
public class ChickensShed {
    public static final String MODID = "chickensshed";

    public static ConfigHandler CONFIG = new ConfigHandler();

    public ChickensShed(){
        FMLJavaModLoadingContext.get().getModEventBus().addListener(this::preInitClient);
        AutoConfig.register(ConfigHandler.class, JanksonConfigSerializer::new);
        CONFIG = AutoConfig.getConfigHolder(ConfigHandler.class).getConfig();
        MinecraftForge.EVENT_BUS.register(this);
    }

    private void preInitClient(final FMLClientSetupEvent event){
        GUIConfigHandler.init();
    }

    @SubscribeEvent
    public void onLivingUpdate(LivingEvent.LivingTickEvent event) {
        LivingEntity entity = event.getEntity();
        if (!entity.level.isClientSide && entity instanceof Chicken chicken) {
            if (chicken.isBaby() && !CONFIG.chicksDropFeathers) {
                return;
            }
            if (chicken.level.random.nextInt(CONFIG.dropFrequency) == 0) {
                chicken.spawnAtLocation(Items.FEATHER, 1);
            }
        }
    }
}

File: src/main/kotlin/com/holmraven/chickensshed/ChickensShed.kt
package com.holmraven.chickensshed

import net.minecraftforge.fml.common.Mod
import thedarkcolour.kotlinforforge.forge.FORGE_BUS
import thedarkcolour.kotlinforforge.forge.MOD_BUS
import net.minecraftforge.eventbus.api.SubscribeEvent
import net.minecraftforge.event.entity.living.LivingEvent
import net.minecraft.entity.passive.ChickenEntity
import net.minecraft.item.Items

@Mod(ChickensShed.ID)
object ChickensShed {
    const val ID: String = "chickensshed"

    init {
        FORGE_BUS.register(this)
        MOD_BUS.register(this)
    }

    @SubscribeEvent
    fun onLivingUpdate(event: LivingEvent.LivingUpdateEvent) {
        if (event.entity.world.isRemote || event.entity !is ChickenEntity) return
        val chicken: ChickenEntity = event.entity as ChickenEntity
        if (chicken.world.rand.nextInt(26000) == 0) chicken.entityDropItem(Items.FEATHER, 1)
    }
}